plugins/modules/postgresql_ext.py

```diff
@@ -272,60 +291,71 @@ def ext_get_versions(cursor, ext):
       ext (str) -- extension name
     """
 
+    current_version = None
+    params = {}
+    params['ext'] = ext
+
     # 1. Get the current extension version:
     query = ("SELECT extversion FROM pg_catalog.pg_extension "
              "WHERE extname = %(ext)s")
 
-    current_version = None
-    cursor.execute(query, {'ext': ext})
+    cursor.execute(query, params)
+
     res = cursor.fetchone()
     if res:
         current_version = res[0]
 
     # 2. Get available versions:
     query = ("SELECT version FROM pg_available_extension_versions "
              "WHERE name = %(ext)s")
-    cursor.execute(query, {'ext': ext})
-    res = cursor.fetchall()
 
-    available_versions = parse_ext_versions(current_version, res)
+    cursor.execute(query, params)
+
+    available_versions = [r[0] for r in cursor.fetchall()]
 
     if current_version is None:
         current_version = False
 
     return (current_version, available_versions)
 
 
-def parse_ext_versions(current_version, ext_ver_list):
-    """Parse ext versions.
+def ext_valid_update_path(cursor, ext, current_version, version):
+    """
+    Check to see if the installed extension version has a valid update
+    path to the given version. A version of 'latest' is always a valid path.
+
+    Return True if a valid path exists. Otherwise return False.
 
     Args:
-      current_version (str) -- version to compare elements of ext_ver_list with
-      ext_ver_list (list) -- list containing dicts with versions
+      cursor (cursor) -- cursor object of psycopg2 library
+      ext (str) -- extension name
+      current_version (str) -- installed version of the extension.
+      version (str) -- target extension version to update to.
+        A value of 'latest' is always a valid path and will result
+        in the extension update command always being run.
+    """
 
-    Return a sorted list with versions that are higher than current_version.
+    valid_path = False
+    params = {}
+    if version != 'latest':
+        query = ("SELECT path FROM pg_extension_update_paths(%(ext)s) "
+                 "WHERE source = %(cv)s "
+                 "AND target = %(ver)s")
 
-    Note: Incomparable versions (e.g., postgis version "unpackaged") are skipped.
-    """
-    available_versions = []
+        params['ext'] = ext
+        params['cv'] = current_version
+        params['ver'] = version
 
-    for line in ext_ver_list:
-        if line['version'] == 'unpackaged':
-            continue
+        cursor.execute(query, params)
+        executed_queries.append(cursor.mogrify(query, params))
```

Comment:
```suggestion
        executed_queries.append(cursor.mogrify(query, params))
```
I'm not sure we should log this. This field was introduced to return only queries that modify something, not selects, etc.
I'm not strictly opposed the idea but in this case we, at least, should log all the selects.
I would suggest removing logging of this query because for users, i thinks, it can be important to see what has been changed, not all of the queries that the module executes for its internal purposes.
What do you folks think?

Reply:
I'd decided to add it, and not other select statements, because I thought it a bit more informative as to why PostgreSQL may be returning that the update path is provided was invalid.
I'm fine with removing it if you'd like though.

Reply:
yeah, i would suggest removing it as we have the path mentioned in the doc

Reply:
Removed. Updated the docs around the version parameter as well.